dotcom-rendering/src/components/CricketScorecard.tsx

```diff
@@ -0,0 +1,379 @@
+};
+
+type Extras = {
+	byes: number;
+	legByes: number;
+	noBalls: number;
+	penalties: number;
+	wides: number;
+};
+
+type InningsTotals = {
+	runs: number;
+	overs: string;
+	wickets: number;
+	extras: number;
+};
+
+type FallOfWicketData = {
+	order: number;
+	name: string;
+	runs: number;
+};
+
+type CricketTeam = {
+	name: string;
+	lineup: string[];
+};
+
+type InningsData = {
+	description: string;
+	bowlers: BowlerData[];
+	batters: BatterData[];
+	extras: Extras;
+	inningsTotals: InningsTotals;
+	fallOfWickets: FallOfWicketData[];
+};
+
+const Bowling = ({ bowlers }: { bowlers: BowlerData[] }) => (
+	<table css={tableStyles}>
+		<thead>
+			<tr>
+				<th>Bowler</th>
+				<th>O</th>
+				<th>M</th>
+				<th>R</th>
+				<th>W</th>
+			</tr>
+		</thead>
+		<tbody>
+			{bowlers.map((bowler) => (
+				<tr key={bowler.name}>
+					<th scope="row">{bowler.name}</th>
+					<td>
+						{bowler.overs}.{bowler.balls % 6}
+					</td>
+					<td>{bowler.maidens}</td>
+					<td>{bowler.runs}</td>
+					<td>{bowler.wickets}</td>
+				</tr>
+			))}
+		</tbody>
+	</table>
+);
+
+const Batting = ({
+	batters,
+	extras,
+	inningsTotals,
+}: {
+	batters: BatterData[];
+	extras: Extras;
+	inningsTotals: InningsTotals;
+}) => (
+	<table
+		css={[
+			tableStyles,
+			css`
+				${until.leftCol} {
+					border-top: 0.0625rem solid
+						${palette('--football-match-list-border')};
+				}
+			`,
+		]}
+	>
+		<thead>
+			<tr>
+				<th>Batter</th>
+				<th css={hideUntilTabletStyle}></th>
+				<th>Runs</th>
+				<th>Balls</th>
+				<th css={hideUntilTabletStyle}>4s</th>
+				<th css={hideUntilTabletStyle}>6s</th>
+			</tr>
+		</thead>
+		<tbody>
+			{batters.map((batter) => (
+				<tr key={batter.name}>
+					<th scope="row">
+						<strong>{batter.name}</strong>
+						<div css={hideFromTabletStyle}>{batter.howOut}</div>
+					</th>
+					<td css={hideUntilTabletStyle}>{batter.howOut}</td>
+					<td>{batter.runs}</td>
+					<td>{batter.ballsFaced}</td>
+					<td css={hideUntilTabletStyle}>{batter.fours}</td>
+					<td css={hideUntilTabletStyle}>{batter.sixes}</td>
+				</tr>
+			))}
+			<tr
+				css={css`
+					th,
+					td {
+						border-top: 0.0625rem dashed
+							${palette('--cricket-scorecard-divider')};
+					}
+				`}
+			>
+				<th scope="row">
+					<strong>Extras</strong>
+				</th>
+				<td css={hideUntilTabletStyle}>
+					{getExtrasDescription(extras)}
+				</td>
+				<td css={hideUntilTabletStyle} colSpan={4}>
+					{inningsTotals.extras}
+				</td>
+				<td css={hideFromTabletStyle} colSpan={2}>
+					{inningsTotals.extras}
+				</td>
+			</tr>
+			<tr>
+				<th scope="row">
+					<strong>Total</strong>
+				</th>
+				<td css={hideUntilTabletStyle}>
+					<strong>for {inningsTotals.wickets}</strong>
+				</td>
+				<td>
+					<strong>{inningsTotals.runs}</strong>
+				</td>
+				<td colSpan={3} css={hideUntilTabletStyle}>
+					<strong>{inningsTotals.overs} overs</strong>
+				</td>
+				<td css={hideFromTabletStyle}></td>
+			</tr>
```

Comment:
I think these should probably be in a `tfoot` given they have different structure and meaning from the rest of the table? The "Total" row seems to fit into the definition of a `tfoot`[^1], but I'm less sure about what the "Extras" row represents. Are you able to describe what it contains and how it relates to the rest of the table?

[^1]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tfoot

Reply:
Nice. `tfoot` looks perfect for the totals. The extras row represents the extra runs given away by bowling mistakes. So it's not really a summary of the other columns. I'd err on leaving it out of the foot